tests/Integration/ApiPlatform/ModuleEndpointTest.php

```diff
@@ -220,6 +225,46 @@ public function testUpdateModuleStatus(array $module): void
         $this->assertEquals(0, $disabledModules['totalItems']);
     }
 
+    /**
+     * @depends testBulkUpdateStatus
+     */
+    public function testResetModule(array $module): void
+    {
+        // Reset specific module
+        $bearerToken = $this->getBearerToken(['module_read', 'module_write']);
+        $response = static::createClient()->request('PUT', sprintf('/module/%s/reset', $module['technicalName']), [
+            'auth_bearer' => $bearerToken,
+            'json' => [
+                'keepData' => false,
+            ],
+        ]);
+        self::assertResponseStatusCodeSame(200);
+        $decodedResponse = json_decode($response->getContent(), true);
+        $this->assertNotFalse($decodedResponse);
+
+        // Check response from status update request
+        $expectedModuleInfos = [
+            'moduleId' => $module['moduleId'],
+            'technicalName' => $module['technicalName'],
+            'version' => $module['version'],
+            'enabled' => true,
+            'installed' => true,
+        ];
+        $this->assertEquals($expectedModuleInfos, $decodedResponse);
+    }
+
+    public function testResetModule_NotInstalled(): void
+    {
+        $module = [
+            'technicalName' => 'ps_notthere',
+        ];
+        $bearerToken = $this->getBearerToken(['module_read', 'module_write']);
```

Comment:
The use case you aimed at testing is also interesting though, checking that a module is disabled or uninstalled and try to reset it which will throw a dedicated exception (that you already implemented) because the action is not allowed It makes sense but to have the proper conditions you will need to change the status of the module first: disable it using the status API, and uninstall it using the uninstall API

But since the uninstall is not available yet you can only assert the behaviour for disabled modules for now, the uninstalled case can be added later